GVFS/GVFS.Platform.Windows/ProjFSFilter.cs

```diff
@@ -679,9 +679,16 @@ private static bool TryEnableProjFSOptionalFeature(ITracer tracer, PhysicalFileS
 
         private static ProcessResult GetProjFSOptionalFeatureStatus()
         {
-            return CallPowershellCommand(
-                "$var=(Get-WindowsOptionalFeature -Online -FeatureName " + OptionalFeatureName + ");  if($var -eq $null){exit " +
-                (int)ProjFSInboxStatus.NotInbox + "}else{if($var.State -eq 'Enabled'){exit " + (int)ProjFSInboxStatus.Enabled + "}else{exit " + (int)ProjFSInboxStatus.Disabled + "}}");
+            try
+            {
+                return CallPowershellCommand(
+                    "$var=(Get-WindowsOptionalFeature -Online -FeatureName " + OptionalFeatureName + ");  if($var -eq $null){exit " +
+                    (int)ProjFSInboxStatus.NotInbox + "}else{if($var.State -eq 'Enabled'){exit " + (int)ProjFSInboxStatus.Enabled + "}else{exit " + (int)ProjFSInboxStatus.Disabled + "}}");
+            }
+            catch (Exception)
+            {
+                return new ProcessResult(string.Empty, "Failed. Possible that powershell does not exist on system", 1);
+            }

```

Comment:
Some questions to think about.
1. Should this be moved down a level into the `CallPowershellCommand` method so that anything else that calls it will catch it?
2. Should we be using `where` to try and locate powershell and use the full path or error out if not found?
3. Should we fallback on using `dism /online /get-featureinfo /featurename:Client-ProjFS` and trying to parse its output for `State : Enabled`?

Reply:
> Some questions to think about.
> 
> 1. Should this be moved down a level into the `CallPowershellCommand` method so that anything else that calls it will catch it? 

Yes. That would be better. Should have known better.

> 2. Should we be using `where` to try and locate powershell and use the full path or error out if not found?

Good idea. I haven't heard back from the user to see if they just don't have it at all, or if their PATH is messed up.

> 3. Should we fallback on using `dism /online /get-featureinfo /featurename:Client-ProjFS` and trying to parse its output for `State : Enabled`?

Good thing the service runs elevated or else `dism` wouldn't work. Worth a shot.

Reply:
Hm. The fallback is probably more complicated than we want, since the `ProcessResult` is taken and its exit code is converted into an `ProjFSInboxStatus` enum.

We should focus only on failing gracefully when not finding powershell. Using `where` to find it is a really valuable suggestion.